src/Symfony/Bundle/FrameworkBundle/EventListener/ServiceResetSubscriber.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Clean up services between requests.
+ *
+ * @author Alexander M. Turek <me@derrabus.de>
+ *
+ * @internal
+ */
+class ServiceResetSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var \Traversable
+     */
+    private $services;
+
+    /**
+     * @var array
+     */
+    private $resetMethods;
+
+    public function __construct(\Traversable $services, array $resetMethods)
+    {
+        $this->services = $services;
+        $this->resetMethods = $resetMethods;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            KernelEvents::TERMINATE => array('onKernelTerminate', -2048),
+        );
+    }
+
+    public function onKernelTerminate()
+    {
+        foreach ($this->services as $id => $service) {
+            foreach ($this->resetMethods[$id] as $method) {
+                call_user_func(array($service, $method));
```

Comment:
`$service->$method()` would be faster by avoid the `call_user_func` frame in the stack